src/librustc_privacy/lib.rs

```diff
@@ -433,15 +442,6 @@ impl<'b, 'a, 'tcx> TypeVisitor<'tcx> for ReachEverythingInTheInterfaceVisitor<'b
 
         ty.super_visit_with(self)
     }
-
-    fn visit_trait_ref(&mut self, trait_ref: ty::TraitRef<'tcx>) -> bool {
-        if let Some(node_id) = self.ev.tcx.hir.as_local_node_id(trait_ref.def_id) {
-            let item = self.ev.tcx.hir.expect_item(node_id);
```

Comment:
There's a `super_visit_ty` call, the `DefId` being looked at is independent of the recursion.